include/deal.II/base/aligned_vector.h

```diff
@@ -486,10 +493,22 @@ AlignedVector<T>::operator = (const AlignedVector<T> &vec)
 template < class T >
 inline
 void
-AlignedVector<T>::resize_fast (const size_type size)
+AlignedVector<T>::resize_fast (const size_type size_in)
 {
-  reserve (size);
-  _end_data = _data + size;
+  const size_type old_size = size();
+  if (std_cxx11::is_trivial<T>::value == false && size_in < old_size)
+    {
+      // call destructor on fields that are released
+      while (_end_data != _data+size_in)
+        (--_end_data)->~T();
```

Comment:
I think the reason was that it was particularly simple that way because end_data has the correct value after the loop. The reverse order was rather a by-product, but I will mention it in the file now.